File: src/main/java/org/jdc/template/ux/about/AboutActivity.java
package org.jdc.template.ux.about;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.text.format.DateUtils;
import android.view.MenuItem;
import android.widget.TextView;
import android.widget.Toast;

import org.jdc.template.BuildConfig;
import org.jdc.template.R;
import org.jdc.template.inject.Injector;
import org.jdc.template.ui.activity.BaseActivity;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class AboutActivity extends BaseActivity implements AboutContract.View {
    @BindView(R.id.versionTextView)
    TextView versionTextView;
    @BindView(R.id.versionDateTextView)
    TextView versionDateTextView;
    @BindView(R.id.mainToolbar)
    Toolbar toolbar;

    @Inject
    AboutPresenter presenter;

    public AboutActivity() {
        Injector.get().inject(this);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        enableActionBarBackArrow();

        versionTextView.setText(BuildConfig.VERSION_NAME + " (" + BuildConfig.VERSION_CODE + ")");
        versionDateTextView.setText(DateUtils.formatDateTime(this, BuildConfig.BUILD_TIME, DateUtils.FORMAT_SHOW_DATE | DateUtils.FORMAT_SHOW_TIME | DateUtils.FORMAT_SHOW_YEAR));

        presenter.init(this);
        presenter.load();
    }

    @Override
    protected void onStart() {
        super.onStart();
        presenter.register();
    }

    @Override
    protected void onStop() {
        presenter.unregister();
        super.onStop();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    @OnClick(R.id.createDatabaseButton)
    public void onCreateDatabaseButtonClick() {
        presenter.createSampleData();
    }

    @OnClick(R.id.jobTestButton)
    public void onJobTestClick() {
        presenter.jobTest();
    }

    @OnClick(R.id.restTestButton)
    public void onRestTestButtonClick() {
        presenter.testQueryWebServiceCallRx();
    }

    @OnClick(R.id.rxTestButton)
    public void onTestRxClick() {
        presenter.testRx();
    }

    @Override
    public void showMessage(String message) {
        Toast.makeText(this, message, Toast.LENGTH_LONG).show();
    }

}

File: src/main/kotlin/org/jdc/template/ux/about/AboutActivity.kt
package org.jdc.template.ux.about

import android.arch.lifecycle.ViewModelProvider
import android.arch.lifecycle.ViewModelProviders
import android.content.Intent
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.google.android.gms.oss.licenses.OssLicensesMenuActivity
import me.eugeniomarletti.extras.ActivityCompanion
import org.jdc.template.R
import org.jdc.template.databinding.AboutActivityBinding
import org.jdc.template.inject.Injector
import org.jdc.template.ui.activity.BaseActivity
import javax.inject.Inject

class AboutActivity : BaseActivity() {

    @Inject
    lateinit var viewModelFactory: ViewModelProvider.Factory

    private val viewModel by lazy { ViewModelProviders.of(this, viewModelFactory).get(AboutViewModel::class.java) }
    private val binding by lazy { DataBindingUtil.setContentView<AboutActivityBinding>(this, R.layout.about_activity) }

    init {
        Injector.get().inject(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.apply {
            viewModel = this@AboutActivity.viewModel
            setLifecycleOwner(this@AboutActivity)
        }

        viewModel.logAnalytics()

        setSupportActionBar(binding.appbar.mainToolbar)
        enableActionBarBackArrow(true)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_about, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.menu_item_licenses -> {
                OssLicensesMenuActivity.setActivityTitle(getString(R.string.menu_lib_licenses))
                startActivity(Intent(this, OssLicensesMenuActivity::class.java))
            }
            else -> return super.onOptionsItemSelected(item)
        }

        return true
    }

    companion object : ActivityCompanion<IntentOptions>(IntentOptions, AboutActivity::class)

    object IntentOptions
}
